python/cudf/cudf/core/dataframe.py

```diff
@@ -3421,11 +3421,10 @@ def as_gpu_matrix(self, columns=None, order="F"):
         ):
             raise TypeError("non-numeric data not yet supported")
 
-        if all([is_datetime_dtype(col.dtype) for col in cols]):
-            # np bug: https://github.com/numpy/numpy/issues/17428
-            dtype = np.dtype("datetime64[ms]")
-        else:
-            dtype = np.find_common_type(cols, [])
+        try:
+            dtype = np.result_type(*cols)
```

Comment:
Does this work via ducktyping? Should we just pass in the dtypes instead?

Reply:
I'm not sure if I understand. Could you elaborate a bit more?

Reply:
`as_gpu_matrix` supports taking a subset of columns to construct the cupy matrix. Do you mean something like:
```
dtypes = [self.dtypes[t] for t in columns]
dtype = np.result_type(*dtypes)
```

Reply:
Yea, right now we're passing `Column` objects directly into the `np.result_type` function which I'd expect to error as numpy doesn't understand our Classes. I'd expect something like:
```
dtype = np.result_type(*(col.dtype for col in cols))
```